TwoNote/app/src/main/java/com/microsoft/device/display/samples/twonote/NoteFragment.kt

```diff
@@ -0,0 +1,120 @@
+        val blueButton = view.findViewById<Button>(R.id.button_blue)
+        blueButton.setOnClickListener { chooseColor(PaintColors.Blue.name) }
+        val greenButton = view.findViewById<Button>(R.id.button_green)
+        greenButton.setOnClickListener { chooseColor(PaintColors.Green.name) }
+        val yellowButton = view.findViewById<Button>(R.id.button_yellow)
+        yellowButton.setOnClickListener { chooseColor(PaintColors.Yellow.name) }
+        val purpleButton = view.findViewById<Button>(R.id.button_purple)
+        purpleButton.setOnClickListener { chooseColor(PaintColors.Purple.name) }
+
+        recoverDrawing()
+
+        drawView.viewTreeObserver.addOnDrawListener {
+            copyDrawBitmapIfAdded()
+        }
+        drawView.setPaintRadius(0)
+
+        return view
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        drawView.viewTreeObserver.removeOnDrawListener { }
+    }
+
+    // Drawing related
+    private fun chooseColor(title: String) {
+        when (title) {
+            PaintColors.Red.name -> drawView.changePaintColor(ContextCompat.getColor(requireActivity().applicationContext, R.color.red))
+            PaintColors.Blue.name -> drawView.changePaintColor(ContextCompat.getColor(requireActivity().applicationContext, R.color.blue))
+            PaintColors.Green.name -> drawView.changePaintColor(ContextCompat.getColor(requireActivity().applicationContext, R.color.green))
+            PaintColors.Yellow.name -> drawView.changePaintColor(ContextCompat.getColor(requireActivity().applicationContext, R.color.yellow))
+            PaintColors.Purple.name -> drawView.changePaintColor(ContextCompat.getColor(requireActivity().applicationContext, R.color.purple))
+        }
+    }
+
+    private fun recoverDrawing() {
+        val viewModel = ViewModelProvider(requireActivity()).get(DrawViewModel::class.java)
+        val strokes = viewModel.getStrokeList()
+        drawView.setStrokeList(strokes)
+        val angle = viewModel.getPenRadius()
+        drawView.setPaintRadius(angle)
+    }
+
+    private fun clearDrawing() {
+        drawView.clearDrawing()
+        val viewModel = ViewModelProvider(requireActivity()).get(DrawViewModel::class.java)
+        viewModel.setImageLiveData(null)
+        viewModel.setStrokeList(listOf())
+    }
+
+    private fun copyDrawBitmap() {
+        val viewModel = ViewModelProvider(requireActivity()).get(DrawViewModel::class.java)
+        val drawBitmap = drawView.getDrawBitmap()
+        if (drawBitmap != null) {
+            viewModel.setImageLiveData(drawBitmap)
+        }
+        val strokeList = drawView.getStrokeList()
+        if (strokeList.isNotEmpty()) {
+            viewModel.setStrokeList(strokeList)
+        }
+        //viewModel.setPenRadius(150)
+    }
+
+    private fun copyDrawBitmapIfAdded() {
+        if (isAdded) {
+            copyDrawBitmap()
+        }
+    }
+}
```

Comment:
this is just linting stuff haha but i think we're supposed to have newlines at the end of files?

idk if you've tried before but you can lint by running "gradlew lint" or "gradlew ktlint" in the terminal in android studio